contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```diff
@@ -0,0 +1,415 @@
+
+  /**
+   * Comma separated list mapping a field in Avro schema to POJO field eg :
+   * orderId:orderId:INTEGER
+   */
+  public void setGenericRecordToPOJOFieldsMapping(String genericRecordToPOJOFieldsMapping)
+  {
+    this.genericRecordToPOJOFieldsMapping = genericRecordToPOJOFieldsMapping;
+  }
+
+  @InputPortFieldAnnotation(optional = false)
+  public final transient DefaultInputPort<GenericRecord> data = new DefaultInputPort<GenericRecord>()
+  {
+    @Override
+    public void process(GenericRecord tuple)
+    {
+      processTuple(tuple);
+    }
+  };
+
+  /**
+   * Converts given Generic Record and to a POJO and emits it
+   */
+  protected void processTuple(GenericRecord tuple)
+  {
+    try {
+      Object obj = getPOJOFromGenericRecord(tuple, getCls());
+
+      if (obj != null) {
+        output.emit(obj);
+        recordCnt++;
+      }
+
+    } catch (InstantiationException | IllegalAccessException e) {
+      LOG.error("Could not initialize object of class - " + getClass().getName());
+      errorCnt++;
+    }
+  }
+
+  /**
+   * Returns a POJO from a Generic Record
+   * 
+   * @return Object
+   */
+  @SuppressWarnings("unchecked")
+  private Object getPOJOFromGenericRecord(GenericRecord tuple, Class<?> cls)
+      throws InstantiationException, IllegalAccessException
+  {
+    Object newObj = getCls().newInstance();
+
+    try {
+      for (int i = 0; i < columnFieldSetters.size(); i++) {
+
+        AvroToPojo.ActiveFieldInfo afi = columnFieldSetters.get(i);
+        afi = columnFieldSetters.get(i);
+        SupportType st = afi.fieldInfo.getType();
+        Object val = null;
+
+        try {
+          val = tuple.get(afi.fieldInfo.getColumnName());
+        } catch (Exception e) {
+          LOG.error("Could not find field -" + afi.fieldInfo.getColumnName() + "- in the generic record");
+          val = null;
+          fieldErrorCnt++;
+        }
+
+        if (val == null) {
+          continue;
+        }
+
+        try {
+          switch (st) {
+            case BOOLEAN:
+              ((PojoUtils.SetterBoolean<Object>)afi.setterOrGetter).set(newObj,
+                  (boolean)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case DOUBLE:
+              ((PojoUtils.SetterDouble<Object>)afi.setterOrGetter).set(newObj,
+                  (double)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case FLOAT:
+              ((PojoUtils.SetterFloat<Object>)afi.setterOrGetter).set(newObj,
+                  (float)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case INTEGER:
+              ((PojoUtils.SetterInt<Object>)afi.setterOrGetter).set(newObj,
+                  (int)tuple.get(afi.fieldInfo.getColumnName()));
+              break;
+
+            case STRING:
+              ((PojoUtils.Setter<Object, String>)afi.setterOrGetter).set(newObj,
+                  new String(tuple.get(afi.fieldInfo.getColumnName()).toString()));
```

Comment:
Ok then new String() is not required as toString() returns String object itself.